### PR TITLE
Add MBC tests

### PR DESCRIPTION
**Pull request by oyagci**

### CODE REVIEW

File: src/MemoryBankController1.hpp [hunk 1]
```diff
@@ -33,6 +28,8 @@ class MemoryBankController1 : public AMemoryBankController {
     ramData = ramPtr;
   }
 
+  friend class Accessor;
```
Comment (Alaric1001) [inline src/MemoryBankController1.hpp]: Is this truly needed ? It seems that you just call public methods in your tests.

File: tests/MemoryBankController1.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+public:
+  Accessor() : romData(), ramData(), mbc(&romData, &ramData) {}
+
+  static constexpr auto RAMSize = AMemoryBankController::RAMSize;
+};
+
+class mbc1_f : public ::testing::Test {
+public:
+  Accessor accessor;
```
Comment (Alaric1001) [inline tests/MemoryBankController1.cpp]: You could just declare the 3 Accessor's attributes in here.

File: tests/MemoryBankController1.cpp [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+TEST_F(mbc1_f, rom_data_bank_switch) {
+  char c;
+  int i;
+
+  accessor.mbc.write(0x6000, 0x0);
+  accessor.mbc.write(0x2000, 0x0);
+  accessor.mbc.write(0x4000, 0x0);
+
+  accessor.romData.reserve(0xFFFFF);
```
Comment (Alaric1001) [inline tests/MemoryBankController1.cpp]: Did you mean `resize()` ?

File: src/MemoryBankController1.hpp (hunk 1 above)
Comment (oyagci) [reply src/MemoryBankController1.hpp]: indeed

**Pull request merged**